addons/hr_work_entry_holidays/models/hr_leave.py

```diff
@@ -239,11 +239,12 @@ def _get_number_of_days(self, date_from, date_to, employee):
             # Use sudo otherwise base users can't compute number of days
             contracts = employee.sudo()._get_contracts(date_from, date_to, states=['open', 'close'])
             contracts |= employee.sudo()._get_incoming_contracts(date_from, date_to)
-            calendar = contracts[:1].resource_calendar_id if contracts else None # Note: if len(contracts)>1, the leave creation will crash because of unicity constaint
+            calendar = contracts[:1].resource_calendar_id if contracts else self.env.company.resource_calendar_id # Note: if len(contracts)>1, the leave creation will crash because of unicity constaint
```

Comment:
```suggestion
            calendar = contracts[:1].resource_calendar_id if contracts else employee.company_id.resource_calendar_id
            # Note: if len(contracts)>1, the leave creation will crash because of unicity constaint
```